### PR TITLE
ARGO-48 Add logging to the data model classes

### PR DESCRIPTION
**Pull request by kaggis**
Add logging mechanism to Data Model classes under .ops package

- Add log4j pom dependency
- Add log4j Logger object on classes that load resources (configuration manager, status operations manager)
- Handle and log exceptions when loading resources (json files)

### CODE REVIEW

**Comment by kaggis** [comment]
please review
/cc @skanct @dpavlos

File: status-computation/java/src/main/java/ops/ConfigManager.java [hunk 1]
```diff
@@ -108,6 +123,18 @@ public void loadJson(File json_file) throws FileNotFoundException {
 			}
 		}
```
Comment (skanct) [inline status-computation/java/src/main/java/ops/ConfigManager.java]: I think here the alignment is off

File: status-computation/java/src/main/java/ops/ConfigManager.java [hunk 2]
```diff
@@ -48,67 +58,80 @@ public void clear() {
...
+			JsonObject jEgroupTags = jObj.getAsJsonObject("egroup_tags");
+			JsonObject jGgroupTags = jObj.getAsJsonObject("ggroup_tags");
+			JsonObject jMdataTags = jObj.getAsJsonObject("mdata_tags");
+			JsonObject jDataMap = jObj.getAsJsonObject("datastore_maps");
+
+			// Get compound fields
+			for (Entry<String, JsonElement> item : jEgroupTags.entrySet()) {
+
+				this.egroupTags.put(item.getKey(), item.getValue()
+						.getAsString());
```
Comment (dpavlos) [inline status-computation/java/src/main/java/ops/ConfigManager.java]: check spaces

File: status-computation/java/src/main/java/ops/OpsManager.java [hunk 3]
```diff
@@ -4,230 +4,235 @@
...
+import com.google.gson.JsonParseException;
 import com.google.gson.JsonParser;
 
 public class OpsManager {
-	
-	private HashMap<String,Integer> states;
-	private HashMap<String,Integer> ops;
+
+	private static final Logger LOG = Logger.getLogger(OpsManager.class
+			.getName());
```
Comment (dpavlos) [inline status-computation/java/src/main/java/ops/OpsManager.java]: check new line

File: status-computation/java/src/main/java/ops/OpsManager.java [hunk 4]
```diff
@@ -4,230 +4,235 @@
...
+		BufferedReader br = null;
+		try {
+			br = new BufferedReader(new FileReader(jsonFile));
+
+			JsonParser json_parser = new JsonParser();
+			JsonElement j_element = json_parser.parse(br);
+			JsonObject j_obj = j_element.getAsJsonObject();
+			JsonArray j_states = j_obj.getAsJsonArray("states");
+			JsonObject j_ops = j_obj.getAsJsonObject("operations");
+			this.defaultMissingState = j_obj.getAsJsonPrimitive(
```
Comment (dpavlos) [inline status-computation/java/src/main/java/ops/OpsManager.java]: check new line

File: status-computation/java/src/main/java/ops/OpsManager.java [hunk 5]
```diff
@@ -4,230 +4,235 @@
...
 		this.revStates = new ArrayList<String>();
 		this.revOps = new ArrayList<String>();
-		
+
 		this.truthTable = null;
-		
-		this.order = false;;
+
+		this.order = false;
+		;
```
Comment (dpavlos) [inline status-computation/java/src/main/java/ops/OpsManager.java]: redundant ";" character

**Comment by dpavlos** [comment]
+1

**Pull request merged**